### ISSUE TITLE
Accessibility: Accessible name for button not descriptive enough

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52089]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

This button's name, as read by screen readers, does not describe its purpose.   The button only says "Apply Filter" but does not specify which filter is being applied. If multiple filters are on the page, there are multiple of these buttons which cannot be distinguished though they have different functionality.

![Image](https://github.com/user-attachments/assets/130610ac-9b47-4e6d-90e1-527ec52a9295)

### Component or Page: 

Component: Filter by attribute block

### Name affected element:

Filter by attribute block: Apply button(s)

### Relevant code snippets:

`<div class="wc-block-attribute-filter__actions"><button type="submit" class="wp-block-button__link wc-block-filter-submit-button wc-block-components-filter-submit-button wc-block-attribute-filter__button"><span aria-hidden="true">Apply</span><span class="screen-reader-text">Apply filter</span></button></div>`

### Recommended fix:

Change the screen reader text on the button to "Apply filter: $filter_name" so it's clear which filter is being submitted when the button is pressed.
